client/package.json

```diff
@@ -6,8 +6,11 @@
     "prop-types": "^15.7.2",
     "react": "^16.9.0",
     "react-dom": "^16.9.0",
+    "react-redux": "^7.1.0",
```

Comment:
WHy not react-context and remove react-redux, redux, redux-thunk?

Reply:
React context with react.useReducer hook will work the same.

Reply:
@nimish-gupta  I personally prefer using redux over context with useReducer for large applications. Redux has a global state container where as later has component co-located state container. there is nothing like combineReducer there and also for every reducer we need to create separate dispatch unlike Redux which provides one dispatch function that consumes any action dedicated for any reducer function. I like the centralized control flow of state transitions which can be achieved through flux, in this case redux.
I might use context if need to pass a prop down 2-3 levels and it's not a global state or affecting any other component.
Also hardest part about redux is the setup, which is already done. So I am not sure or comfortable in using context and useReducer in this project.
opinions @piyushagwl007 @karansapolia ?

Reply:
> Redux has a global state container where as later has component co-located state container.
Can you elaborate this, please?

> also for every reducer we need to create separate dispatch unlike Redux which provides one dispatch function that consumes any action dedicated for any reducer function.

You can create a context at the top most component and use that dispatch with all the functions/actions.

> the centralized control flow of state transitions which can be achieved through flux

this can be also done with context, again if we create the top most context.
> there is nothing like combineReducer

you can easily create a combined reducer using redux, you just have to call the multiple dispatches at one place.

> I might use context if need to pass a prop down 2-3 levels

This is one of the use cases of redux, but it can also be used for multiple other cases.

With context, you are reducing the complexity of adding multiple new libraries (redux, react-redux, thunk) which will make it difficult to debug in case of a problem, also for debugging you also have to add redux debugging tools for checking redux state in the console. In case of react context, you can use react dev tools only.
I don't have any issue with redux, in the end it will be your team decision. Check this article https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367 (might :))

Reply:
@nimish-gupta after discussing with the team we understood that our usecase can be implemented using both of thhe options but we wanted to learn redux with react so we will be going with redux :)